pkg/aws/eni/routing/routing.go

```diff
@@ -62,6 +63,10 @@ type RoutingInfo struct {
 // mtu: The ENI device MTU.
 // masq: Whether masquerading is enabled.
 func Install(ip net.IP, info *RoutingInfo, mtu int, masq bool) error {
+
+	if (option.Config.IPAM == option.IPAMENI) && (ip.To4() == nil) {
```

Comment:
I think we can safely drop the check on IPAM, and just keep the IP check because this code is only used in ENI mode anyway. Validating whether the input to this function is IPv4 is more valuable. We should also log a warning  level before bailing out.

Reply:
Furthermore, I think this code will become more generic over time as https://github.com/cilium/cilium/pull/11268 was merged recently, bringing Azure into the picture. FWIW, there's a comment in that PR which mentions pulling this package out to a more generic path. (But that doesn't need to be done in this PR.)

Reply:
Also, shouldn't this be returning an actual error? Silently discarding this `Install()` call because it was called wrong seems like it could lead to surprises later on when things just don't work and there's no error to tell you why.

Reply:
Good point.